lib/teleterm/clusters/storage.go

```diff
@@ -159,6 +160,7 @@ func (s *Storage) addCluster(ctx context.Context, dir, webProxyAddress string) (
 		return nil, trace.Wrap(err)
 	}
 
+	clusterClient.TLSRoutingConnUpgradeRequired = apiclient.IsALPNConnUpgradeRequired(webProxyAddress, s.InsecureSkipVerify)
```

Comment:
`TLSRoutingConnUpgradeRequired` shouldn't have any impact on `clusterClient.Ping`, right?

I was wondering if we should set `TLSRoutingConnUpgradeRequired` directly on `cfg` on line 148 before we ping the cluster.

But I suppose with the current approach if we ping a server and it turns out it's not reachable, we avoid making an unnecessary request to it with `IsALPNConnUpgradeRequired`.

Reply:
> But I suppose with the current approach if we ping a server and it turns out it's not reachable, we avoid making an unnecessary request to it with IsALPNConnUpgradeRequired.

yes, I think it's better to do the ping first. Will add a comment here.